File: src/main/java/com/example/cqrs/command/dto/DepositRequest.java
package com.example.cqrs.command.dto;

import lombok.*;

@AllArgsConstructor(access = AccessLevel.PRIVATE)
@NoArgsConstructor(access = AccessLevel.PRIVATE)
@Builder
@Getter
public class DepositRequest {

    private String accountId;
    private double amount;
    private String userId;

    // factory method
    public static DepositRequest of(String accountId, double amount, String userId) {
        return new DepositRequest(accountId, amount, userId);
    }

}

File: src/main/java/com/example/cqrs/command/dto/DepositRequest.kt
package com.example.cqrs.command.dto

data class DepositRequest(
    val accountId: String,
    val amount: Double,
    val userId: String
) {
    companion object {
        fun of(accountId: String, amount: Double, userId: String): DepositRequest {
            return DepositRequest(accountId, amount, userId)
        }
    }
}
